community/server/src/main/java/org/neo4j/server/rest/transactional/Neo4jJsonCodec.java

```diff
@@ -118,14 +128,53 @@ private void writePath( JsonGenerator out, Iterator<PropertyContainer> value ) t
         }
     }
 
-    private void writePropertyContainer( JsonGenerator out, PropertyContainer value ) throws IOException
+    private void writePropertyContainer( JsonGenerator out, PropertyContainer value, TransactionStateChecker txStateChecker )
+            throws IOException
+    {
+        if ( value instanceof Node )
+        {
+            writeNode( out, (Node) value, txStateChecker );
+        }
+        else if ( value instanceof Relationship )
+        {
+            writeRelationship( out, (Relationship) value, txStateChecker );
+        }
+        else
+        {
+            throw new IllegalArgumentException( "Expected a Node or Relationship, but got a " + value.toString() );
+        }
+    }
+
+    private void writeNode( JsonGenerator out, Node node, TransactionStateChecker txStateChecker ) throws IOException
+    {
+        out.writeStartObject();
+        try
+        {
+            if ( !txStateChecker.isNodeDeletedInCurrentTx( node.getId() ) )
+            {
+                for ( Map.Entry<String,Object> property : node.getAllProperties().entrySet() )
+                {
+                    out.writeObjectField( property.getKey(), property.getValue() );
+                }
+            }
+        }
+        finally
+        {
+            out.writeEndObject();
+        }
+    }
+
+    private void writeRelationship( JsonGenerator out, Relationship node, TransactionStateChecker txStateChecker ) throws IOException
```

Comment:
This thing is so copy-pasted that the variable is still called `node`. Couldn't extract the common stuff between this and `writeNode`